components/LocationInput.tsx

```diff
@@ -184,14 +223,22 @@ const LocationInput: FC<LocationInputProps> = ({
     zoomLevelThreshold,
   ]);
 
+  setIsSubmissionEnabled(isInUK && isPastZoomThreshold);
```

Comment:
Maybe put this in a use effect?
```suggestion
useEffect(() => {
    setIsSubmissionEnabled(isInUK && isPastZoomThreshold);
}, [isInUK, isPastZoomThreshold]);
```

Reply:
great catch, tysm!